src/extensions/electrodegeometry/ElectrodeGeometryWidget/ElectrodeGeometryWidget.tsx

```diff
@@ -1,34 +1,8 @@
-import { norm } from 'mathjs'
 import React from "react"
-import { funcToTransform } from '../../common/CanvasWidget'
-import { CanvasPainter } from '../../common/CanvasWidget/CanvasPainter'
 import CanvasWidget from '../../common/CanvasWidget/CanvasWidget'
-import { CanvasDragEvent, CanvasWidgetLayer, ClickEvent, ClickEventType, DiscreteMouseEventHandler, DragHandler, useLayer, useLayers } from "../../common/CanvasWidget/CanvasWidgetLayer"
-import { getBoundingBoxForEllipse, getHeight, getWidth, pointIsInEllipse, RectangularRegion, rectangularRegionsIntersect, transformDistance, Vec2 } from '../../common/CanvasWidget/Geometry'
-
-
-// - allows drag and click selection of electrodes
-// - highlights mouseover of electrodes
-
-enum Color {
-    BASE = 'rgb(0, 0, 255)',
-    SELECTED = 'rgb(196, 196, 128)',
-    HOVER = 'rgb(128, 128, 255)',
-    SELECTEDHOVER = 'rgb(200, 200, 196)',
-    DRAGGED = 'rgb(0, 0, 196)',
-    DRAGGEDSELECTED = 'rgb(180, 180, 150)',
-    DRAGRECT = 'rgba(196, 196, 196, 0.5)'
-}
-enum TextColor {
-    LIGHT = 'rgb(228, 228, 228)',
-    DARK = 'rgb(32, 32, 32)'
-}
-
-export type Electrode = {
-    label: string,
-    x: number,
-    y: number
-}
+import { useLayer, useLayers } from "../../common/CanvasWidget/CanvasWidgetLayer"
```

Comment:
Separate out the electrodeGeometryLayer into its own .ts file so it can be reused.